spec/rest_in_peace/template_sanitizer_spec.rb

```diff
@@ -120,12 +120,20 @@
 
       context 'param which need to be encoded' do
         let(:params) { { id: 'this param need to be encoded' } }
-        let(:encoded_param) { 'this+param+need+to+be+encoded' }
+        let(:encoded_param) { 'this%20param%20need%20to%20be%20encoded' }
```

Comment:
So, you're saying:

* **1:** `/path/with a space/`
  * should become `/path/with%20a%20space/`
  * but it's currently `/path/with+a+space/`
* **2:** `/path/?query=with a space`
  * should become `/path/?query=with+a+space`
  * which is currently correct
* **3:** `/path/with a space/?query=with a space`
  * should become `/path/with%20a%20space/?query=with+a+space`
  * but it's currently `/path/with+a+space/?query=with+a+space`

Right?

After further consideration, this seems legit. My arguments above seem to only apply to the 'query' part, but not the 'path' part.

Can you check if case 3 is working as expected as well?